fontir/src/variations.rs

```diff
@@ -0,0 +1,632 @@
+                known_axes.push(0x10000);
+                axis_name
+            })
+            .cloned()
+            .collect();
+        unordered_axes.sort();
+        ordered_axes.extend(unordered_axes);
+
+        let axis_value_signs = ordered_axes
+            .iter()
+            .map(|axis_name| {
+                location
+                    .get(axis_name)
+                    .map(|coord| {
+                        let value = coord.into_inner();
+                        if value > ZERO {
+                            1i8
+                        } else if value < ZERO {
+                            -1i8
+                        } else {
+                            0i8
+                        }
+                    })
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        let axis_value_abs = ordered_axes
+            .iter()
+            .map(|axis_name| {
+                location
+                    .get(axis_name)
+                    .map(|coord| OrderedFloat(coord.into_inner().abs()))
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        return LocationSortKey {
+            rank,
+            on_axis_points,
+            known_axes,
+            ordered_axes,
+            axis_value_signs,
+            axis_value_abs,
+        };
+    }
+}
+
+/// Sort key for a location.
+///
+/// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L326>
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct LocationSortKey {
+    // 1 for every non-zero entry in the map
+    rank: usize,
+    // -1 for every axis:value entry in location where a location exists whose only non-zero
+    // position is on axis at value. That is, a location exists that is directly on the axis there.
+    on_axis_points: i16,
+    // index in user-specified axis order, otherwise score 0x10000 (65536)
+    known_axes: Vec<usize>,
+    // ordered list of axis in location
+    ordered_axes: Vec<String>,
+    // signs of positions on axes in ordered_axes order
+    axis_value_signs: Vec<i8>,
+    // abs values of positions on axes in ordered_axes order
+    axis_value_abs: Vec<OrderedFloat<f32>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+struct VariationRegion(HashMap<String, Tent>);
+
+impl VariationRegion {
+    fn new() -> Self {
+        VariationRegion(HashMap::new())
+    }
+}
+
+impl VariationRegion {
+    /// The scalar multiplier for the provided location for this region
+    ///
+    /// Returns None for no influence, Some(non-zero value) for influence.
+    ///
+    /// In Python, supportScalar. We only implement the ot=True, extrapolate=False paths.
+    /// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L123>.
+    fn multiplier_at(&self, location: &NormalizedLocation) -> Option<OrderedFloat<f32>> {
```

Comment:
everybody else (including OT spec) calls this a 'scalar', but I presume that 'multiplier' reads clearer to you.

Reply:
I think that's just an error, I should have called it scalar_at. The key change I wanted to make was to move the method onto the region.